internal/appconfig/config.go

```diff
@@ -197,3 +198,18 @@ func (cfg *Config) BuildStrategies() []string {
 
 	return strategies
 }
+
+func (cfg *Config) URL() (*url.URL, error) {
```

Comment:
Adding test cases for this method to internal/appconfig/config_test.go will help a lot discover the corner cases of this function.

You haven't considered apps with `http_service` section defined.

Reply:
Thanks, this is my first time really digging into these types so I appreciate the pointer to the failure cases.

As an aside, does the lint failure give any more context visually? I see there are a couple errors but Actions' UI is notoriously inaccessible and either the raw logs aren't helpful or I'm just not searching for the right keywords.

Reply:
There are 2 linter errors on lines 161 and 163 for file and they are the same:
```
ineffectual assignment to err (ineffassign)
```

I think it means the assignment to the`err` variable will be ignored because it is declared in the block.

Reply:
OK, I believe I've addressed the http_service case. I don't necessarily want to make this perfect since it's literally replacing "http://" + appName + ".fly.dev" but please let me know if I've missed any obvious cases where an app would spin up an HTTP(S) endpoint and not get caught by this.

Also not sure what the right solution for these linter errors should be. I assumed I was just re-assigning the variable and then returning it if it wasn't nil after the switch, but I'll just be super explicit and hope that makes Go happy. :)

Reply:
the http service cases are not complete tho, you need to revisit that logic because it is possible that there is no http service to infer an url from.